adapters/githubclient.go

```diff
@@ -0,0 +1,47 @@
+package adapters
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/go-github/github"
+	"github.com/jenkins-x/jx-release-version/domain"
+	"golang.org/x/oauth2"
+	"net/http"
+	"os"
+)
+
+type GitHubClient struct {
+	client *github.Client
+}
+
+func (g *GitHubClient) ListTags(ctx context.Context, owner string, repo string) ([]domain.Tag, error) {
+	tags, _, err := g.client.Repositories.ListTags(ctx, owner, repo, nil)
+
+	var a []domain.Tag
+
+	for _, e := range tags {
+		a = append(a, domain.Tag{Name: e.GetName()})
+	}
+
+	return a, err
+}
+
+func NewGitHubClient(debug bool) domain.GitClient {
+	var oauth2Client *http.Client
+
+	token := os.Getenv("GITHUB_AUTH_TOKEN")
+	if token != "" {
+		ts := oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: token},
+		)
+		oauth2Client = oauth2.NewClient(context.Background(), ts)
+	} else {
+		if debug {
+			fmt.Println("no GITHUB_AUTH_TOKEN env var found so using unauthenticated request")
+		}
+	}
+
+	return &GitHubClient{
+		client: github.NewClient(oauth2Client),
```

Comment:
The `oauth2Client` is nil when the  token is not set. Does this lead to any issue? I would change the logic to prevent this from happening.

Reply:
As far as I know, the github library needs an oauth2Client just to allow access to private repositories, if oauth2Client is nil and the repository is public the request will work, but if the repository is private will fail with the following error:
```
Received unexpected error: error getting tags: GET https://api.github.com/repos/org/private-repo/tags: 404 Not Found []
```